sm-core-model/src/main/java/com/salesmanager/core/model/catalog/category/image/CategoryImage.java

```diff
@@ -0,0 +1,84 @@
+package com.salesmanager.core.model.catalog.category.image;
+
+import com.salesmanager.core.constants.SchemaConstant;
+import com.salesmanager.core.model.catalog.category.Category;
+import com.salesmanager.core.model.generic.SalesManagerEntity;
+
+import javax.persistence.*;
+import java.io.InputStream;
+import java.util.UUID;
+
+/**
+ * @author Mostafa Saied <mostafa.saied.fci@gmail.com>
+ */
+
+@Entity
+@Table(name = "CATEGORY_IMAGE", schema= SchemaConstant.SALESMANAGER_SCHEMA)
+public class CategoryImage extends SalesManagerEntity<Long, CategoryImage> {
+
+    @Id
+    @Column(name = "CATEGORY_IMAGE_ID")
+    @TableGenerator(name = "TABLE_GEN", table = "SM_SEQUENCER", pkColumnName = "SEQ_NAME", valueColumnName = "SEQ_COUNT", pkColumnValue = "PRODUCT_IMG_SEQ_NEXT_VAL")
+    @GeneratedValue(strategy = GenerationType.TABLE, generator = "TABLE_GEN")
+    private Long id;
+
+
+    @ManyToOne
+    @JoinColumn(name = "CATEGORY_ID", nullable = false)
+    private Category category;
+
+    @Column(name = "CATEGORY_IMAGE")
+    private String categoryImage;
+
+    @Column(name = "DEFAULT_IMAGE")
+    private boolean defaultImage = false;
```

Comment:
if we're creating this from scratch, maybe it'll be better to introduce image type as an enum and it can contain many different types and we can add default as one of them.

Reply:
There is already image type enum but for the storage eg. product, category, file, ...etc
but what's the difference between marking the image as default with a flag or add DEFAULT and What other type, to the database, I think that way is better in implementation.